nuxeo-coldstorage/src/test/java/org/nuxeo/coldstorage/operations/S3RestoreFromColdStorageTest.java

```diff
@@ -0,0 +1,95 @@
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+import java.io.IOException;
+import java.util.List;
+
+import org.junit.Test;
+import org.nuxeo.coldstorage.S3ColdStorageFeature;
+import org.nuxeo.coldstorage.S3TestHelper;
+import org.nuxeo.coldstorage.helpers.ColdStorageHelper;
+import org.nuxeo.coldstorage.thumbnail.DummyThumbnailFactory;
+import org.nuxeo.ecm.automation.OperationContext;
+import org.nuxeo.ecm.automation.OperationException;
+import org.nuxeo.ecm.core.api.Blob;
+import org.nuxeo.ecm.core.api.DocumentModel;
+import org.nuxeo.ecm.core.api.NuxeoException;
+import org.nuxeo.ecm.platform.ec.notification.NotificationConstants;
+import org.nuxeo.runtime.test.runner.Features;
+
+@Features(S3ColdStorageFeature.class)
+public class S3RestoreFromColdStorageTest extends AbstractTestRestoreFromColdStorageOperation {
+
+    protected S3TestHelper s3TestHelper = S3TestHelper.getInstance();
+
+    @Test
+    public void shouldBeRestoredByRetrieve() throws IOException, OperationException {
+        DocumentModel documentModel = createFileDocument(session, true);
+        // first make the move to cold storage
+        moveContentToColdStorage(session, documentModel);
+        // Mock AWS Lifecycle rule
+        s3TestHelper.moveBlobContentToGlacier(session.getDocument(documentModel.getRef()));
+        try (OperationContext context = new OperationContext(session)) {
+            context.setInput(documentModel);
+            DocumentModel updatedDocument = (DocumentModel) automationService.run(context, RestoreFromColdStorage.ID);
+
+            // check document
+            assertTrue(updatedDocument.hasFacet(ColdStorageHelper.COLD_STORAGE_FACET_NAME));
+
+            // check blobs
+            Blob coldStorageContent = (Blob) updatedDocument.getPropertyValue(
+                    ColdStorageHelper.COLD_STORAGE_CONTENT_PROPERTY);
+            Blob fileContent = (Blob) updatedDocument.getPropertyValue(ColdStorageHelper.FILE_CONTENT_PROPERTY);
+            assertEquals(DummyThumbnailFactory.DUMMY_THUMBNAIL_CONTENT, fileContent.getString());
+            assertEquals(FILE_CONTENT, coldStorageContent.getString());
+
+            // should being retrieved
+            assertEquals(Boolean.TRUE,
+                    updatedDocument.getPropertyValue(ColdStorageHelper.COLD_STORAGE_BEING_RETRIEVED_PROPERTY));
+
+            // check the restore notification
+            String username = NotificationConstants.USER_PREFIX + session.getPrincipal().getName();
+            List<String> subscriptions = notificationManager.getSubscriptionsForUserOnDocument(username,
+                    updatedDocument);
+            assertTrue(subscriptions.contains(ColdStorageHelper.COLD_STORAGE_CONTENT_RESTORED_NOTIFICATION_NAME));
+            assertFalse(subscriptions.contains(ColdStorageHelper.COLD_STORAGE_CONTENT_AVAILABLE_NOTIFICATION_NAME));
```

Comment:
Could you refactor this part of the code to new method and it will be used in the two places https://github.com/nuxeo/nuxeo-coldstorage/pull/58/files#diff-edba9e806719db160079c8b599d2271aba18185fcdf04b7d8bf08ab7c438b4e1R81-R93